packages/analytics/src/types/analytics.types.ts

```diff
@@ -0,0 +1,69 @@
+import type {
+  CONSENT_KEYS,
+  LOAD_INTEGRATION_TRACK_TYPE,
+  ON_SET_USER_TRACK_TYPE,
+} from '../utils/constants';
+import type { Integration } from '../integrations';
+import type { IntegrationFactory } from '../integrations/Integration';
+import type trackTypes from '../types/trackTypes';
+
+export type TrackTypesValues = typeof trackTypes[keyof typeof trackTypes];
+
+export type ConsentData = {
+  [key in typeof CONSENT_KEYS[number]]: boolean;
+};
+
+export type ContextData = {
+  library: string;
+  version: string;
+} & Record<string, unknown>;
+
+export type EventProperties = Record<string, unknown>;
+export type EventContextData = Record<string, unknown>;
+export type EventContext = ContextData & { event: Record<string, unknown> };
+
+export type EventData<T> = {
+  type: T;
+  event: T extends typeof LOAD_INTEGRATION_TRACK_TYPE
+    ? typeof LOAD_INTEGRATION_TRACK_TYPE
+    : T extends typeof ON_SET_USER_TRACK_TYPE
+    ? typeof ON_SET_USER_TRACK_TYPE
+    : string;
+  user: unknown;
```

Comment:
Can we use `UserData` type here?

Reply:
Yes, forgot to change that case. In this case it has to be `UserData | null` as the `user` property is populated by the call to `this.user()` method which can return null. Thanks!